### PR TITLE
Features/smc 432 Upgrade to solidity 0.8.18 with optimization

### PR DESCRIPTION
**Pull request by kevinnguyen17**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,8 +29,6 @@
   },
   "dependencies": {
     "@nomiclabs/hardhat-waffle": "^2.0.1",
-    "@openzeppelin/contracts": "^2.4.0",
```
Comment (marc-aurele-besner) [inline package.json:32]: it's missing @openzeppelin/contracts-upgradeable in the dependencies, and there seems to be other issue that prevent me from cloning and compiling the contracts

File: contracts/PrizeLinkedAccountVault.sol [hunk 2]
```diff
@@ -287,8 +300,9 @@ contract PrizeLinkedAccountVault is
                 result[t] = previousDrawParticipants[i];
                 t++;
```
Comment (marc-aurele-besner) [inline contracts/PrizeLinkedAccountVault.sol]: This could be changed for ++t; and wrap in unchecked

File: contracts/PrizeLinkedAccountVault.sol [hunk 3]
```diff
@@ -287,8 +300,9 @@ contract PrizeLinkedAccountVault is
...
             }
+            ++i;
```
Comment (marc-aurele-besner) [inline contracts/PrizeLinkedAccountVault.sol:302]: This could be wrap in unchecked (and all following ++i)

**Review by marc-aurele-besner — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (kevinnguyen17) [reply package.json:32]: it is weird, it is not available in my package.json. Nevermind, will refresh that file again

File: contracts/PrizeLinkedAccountVault.sol (hunk 2 above)
Comment (kevinnguyen17) [reply contracts/PrizeLinkedAccountVault.sol]: it is an external view btw.

Comment (kevinnguyen17) [reply contracts/PrizeLinkedAccountVault.sol]: but I will change, to ++t to make it consistent

File: contracts/abstracts/GluwaPrizeDraw.sol [hunk 4]
```diff
@@ -299,15 +290,12 @@ contract GluwaPrizeDraw is Initializable, Context {
...
-            uint256 issuedTickets;
-            for (
-                uint256 i = 0;
-                i < _drawParticipantTicket[drawTimeStamp][owner_].length;
-                i++
-            ) {
+        uint256 length = _drawParticipantTicket[drawTimeStamp][owner_].length;
+        if (length > 0) {
+            uint96 issuedTickets;
+            for (uint256 i; i < length; ) {
```
Comment (ohyung1287) [inline contracts/abstracts/GluwaPrizeDraw.sol:297]: this part can be warped in unchecked when assign issuedTickets and ++i

File: contracts/PrizeLinkedAccountVault.sol (hunk 2 above)
Comment (marc-aurele-besner) [reply contracts/PrizeLinkedAccountVault.sol]: Oups 😅

**Review by ohyung1287 — APPROVED** [review]

**Review by marc-aurele-besner — APPROVED** [review]

**Pull request merged**